source/simulator/parameters.cc

```diff
@@ -1513,17 +1554,37 @@ namespace aspect
       }
       prm.leave_subsection ();
 
-      AssertThrow (use_locally_conservative_discretization ||
+      AssertThrow ((use_locally_conservative_discretization ||
+                    use_equal_order_interpolation_for_stokes)
+                   ||
                    (stokes_velocity_degree > 1),
                    ExcMessage ("The polynomial degree for the velocity field "
                                "specified in the 'Stokes velocity polynomial degree' "
                                "parameter must be at least 2, unless you are using "
+                               "the 'Use equal order interpolation for Stokes' parameter, "
+                               "or if you are using "
                                "a locally conservative discretization as specified by the "
                                "'Use locally conservative discretization' parameter. "
-                               "This is because in the former case, the pressure element "
+                               "In both of these cases, the polynomial degree used for the "
+                               "velocity may be equal to one."
+                               "\n\n"
+                               "The restriction exists because by default, the pressure element "
                                "is of one degree lower and continuous, and if you selected "
                                "a linear element for the velocity, you'd need a continuous "
-                               "element of degree zero for the pressure, which does not exist."))
+                               "element of degree zero for the pressure, which does not exist. "
+                               "On the other hand, if using equal-order interpolation, "
+                               "choosing the polynomial degree as one yields a Q1-Q1 "
+                               "element; using a locally conservative discretization "
+                               "with polynomial degree of one yields a Q1-P0 "
+                               "element."));
+
+      AssertThrow (! (use_locally_conservative_discretization &&
```

Comment:
would it make sense to do the check that the order is 1 if use_equal_order_interpolation_for_stokes is true here?

Reply:
The test here is this:
```
      AssertThrow (! (use_locally_conservative_discretization &&
                      use_equal_order_interpolation_for_stokes),
                   ExcMessage ("You have tried to use both the 'Use locally "
                               "conservative discretization' and 'Use equal order "
                               "interpolation for Stokes' parameters in the input "
                               "file. However, their use is incompatible: you "
                               "can only select one of the two."));
```
What I want to check is that you can't at the same time select to use the $Q_k x DGP_{k-1}$ element and the $Q_k x Q_k$. These two things just don't go together, independent of the degree $k$.

Reply:
I understand. My question was: can you please additionally add
```
AssertThrow(!use_equal_order_interpolation_for_stokes || stokes_velocity_degree==1,
ExcMessage("you need to use degree=1 if you use equal order interpolation.")
```

Reply:
Nah, you don't. You can use equal order elements also in the form of Q2 x Q2. It just requires a different implementation of the projection in the stabilization term. I haven't implemented that yet, but I have the corresponding assertion in the assembly term. That's where the missing functionality is, and so that's where the assertion should be.

The assertions we have in the current place (in `parameters.cc`) are logical errors, rather than things not implemented.

Reply:
I would prefer not to trigger an exception somewhere deep inside a workstream assembly, but I don't need to start a fight over this.